sources/packages/backend/apps/api/src/services/education-program-offering/education-program-offering.service.ts

```diff
@@ -98,7 +99,23 @@ export class EducationProgramOfferingService extends RecordDataModelService<Educ
     programOffering.creator = { id: userId } as User;
     // When creating a new offering, parent id and the primary id are the same.
     programOffering.parentOffering = programOffering;
-    return this.repo.save(programOffering);
+
+    try {
+      return await this.repo.save(programOffering);
+    } catch (error: unknown) {
+      if (error instanceof QueryFailedError) {
+        const postgresError = error as PostgresDriverError;
+        if (
+          postgresError.constraint ===
+          DatabaseConstraintNames.LocationIDProgramIDOfferingNameStudyDatesYearOfStudyIndex
+        ) {
+          throw new CustomNamedError(
+            "Duplication error. An offering with the same name, year of study, start date and end date was found.",
```

Comment:
@HRAGANBC @CarlyCotton @Joshua-Lakusta Should the Duplication check also include the `Offering Intensity` field as well?

Reply:
@sh16011993 Does it currently (prior to this change) include Intensity?

Reply:
> @sh16011993 Does it currently (prior to this change) include Intensity?

No, it didn't include intensity.